metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -0,0 +1,347 @@
+import json
+import logging
+import re
+import time
+from collections import Iterable
+from dataclasses import dataclass, field
+from typing import Any, Dict, List
+
+from datahub.configuration import ConfigModel
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.source.metadata_common import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.common import AuditStamp
+from datahub.metadata.com.linkedin.pegasus2avro.dataset import (
+    UpstreamClass,
+    UpstreamLineage,
+    UpstreamLineageClass,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    BooleanTypeClass,
+    DateTypeClass,
+    MySqlDDL,
+    NullTypeClass,
+    NumberTypeClass,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringTypeClass,
+)
+from datahub.metadata.schema_classes import DatasetPropertiesClass
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DBTSourceReport(SourceReport):
+    tables_scanned = 0
+    filtered: List[str] = field(default_factory=list)
+
+    def report_dbt_node_ingested(self, table_name: str) -> None:
+        self.tables_scanned += 1
```

Comment:
What's the difference between this and `report_workunit` from the `SourceReport` superclass?

Reply:
I don't think there is any, I kinda cargo culted off the SQL source and didn't look at this very closely.